### ISSUE TITLE
Allow to define the structure of the name file of a pdf bill

### ISSUE THREAD

**Issue by ZeHiro** [cozy-labs/konnectors#85]
Hello,

I am developping two new konnectors, one for amazon one for capitaine train. The current module to download the pdf has a fixed structure for the pdf names, as the previsou konnectors are only meant to downoad recurrent bills. 
The bills of amazon or capitainetrain are obviously not recureent, so the structure of the file name does not fit for these konnectors. I would like to propose a "generic" name structure, that could be tweaked, but I don't kwow how to do this. Any idea?

**Comment by frankrousseau** [comment]
I think date and vendors as prefix are mandatory. The rest could be anything (it's the way I do with my manual workflow is).

**Comment by ZeHiro** [comment]
I agree, but now, the module sets the date to YYYYMM :) The idea would be to allow the konnector developper to set a name structure, ans the module would parse it and apply it.

**Comment by kosssi** [comment]
capitainetrain <3

Where is your PR :dancer:

**Comment by ZeHiro** [comment]
Hmm, not now, still working on it :)

**Comment by frankrousseau** [comment]
So tell me if the following changes could help you (ref: https://github.com/cozy-labs/konnectors/blob/master/server/lib/save_data_and_file.coffee#L12):

* Change date format generate by `save_data_and_file` to YYYYMMDDD.
* Check for a suffix in entry to add something after the default name.

```coffeescript
extension = ".pdf"
extension = entry.extension if entry.extension?

fileName = "#{entry.date.format 'YYYYMM'}_#{suffix}"
fileName += "_#{entry.suffix}" if entry.suffix?
fileName += extension
```

**Comment by ZeHiro** [comment]
I was thinking of something even more generic : 
The 'suffix' parameter would be replaced by a fileNameParameter object : 
fileNameParameter = 
     suffix: 'vendor' #Compulsory
     dateFormat: 'YYYYYMM' #Optional 
     extension: 'pdf' #optional 
     otherParameters: 
         [oderID]      #optional, must be a parameter of the entry object

The implementation would be : 
if fileNameParameter.dateFormat?
      fileName = "#entry.date.format fileNameParameter.dateFormat"
else
     fileName = "#entry.datae.format 'YYYYMM'¨ 

fileName += "_#{entry.suffix} 
if fileNameParameter.otherParameters? 
      for parameter in fileNameParameter.otherParameters
            if entry["#{parameter}]?
                  fileName+= "_entry["#{parameter}]"


extension = ".pdf"
extension = entry.extension if entry.extension?
fileName += extension

**Comment by ZeHiro** [comment]
There could even be a default reaction based on the type of fileNameParameter: 
```coffeescript
if typeof fileNameParameter is String
       #Same fileName as now, the fileNameParameter is just the 'suffix'
else
      # The fileName is computed as on my previous comment

**Comment by frankrousseau** [comment]
Yep I will work on it this week. Currently I'm working on a documentation on how to make konnectors. You can see it in the wiki:

https://github.com/cozy-labs/konnectors/wiki

**Comment by frankrousseau** [comment]
By the way, thank you for the details. I understand better your need.

**Comment by frankrousseau** [comment]
I hope to be able to work on it tomorrow.

**Comment by ZeHiro** [comment]
Can be closed for now, the pr #95 is fine for what i asked. Thankx

**Comment by frankrousseau** [comment]
Great!
